ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java

```diff
@@ -0,0 +1,250 @@
+   */
+  @Override
+  public void updateTtl(MessageWriteSet messageSetToUpdate) throws StoreException {
+    checkStarted();
+  }
+
+  @Override
+  public FindInfo findEntriesSince(FindToken token, long maxTotalSizeOfEntries) throws StoreException {
+    throw new UnsupportedOperationException("Method not supported");
+  }
+
+  @Override
+  public Set<StoreKey> findMissingKeys(List<StoreKey> keys) throws StoreException {
+    checkStarted();
+    // Check existence of keys in cloud container
+    Set<StoreKey> missingKeys = new HashSet<>();
+    try {
+      for (StoreKey key : keys) {
+        BlobId blobId = (BlobId) key;
+        if (cloudDestination.doesBlobExist(blobId) == false) {
+          missingKeys.add(key);
+        }
+      }
+    } catch (Exception ex) {
+      throw new StoreException(ex, StoreErrorCodes.IOError);
+    }
+    return missingKeys;
+  }
+
+  @Override
+  public StoreStats getStoreStats() {
+    return null;
+  }
+
+  @Override
+  public boolean isKeyDeleted(StoreKey key) throws StoreException {
+    checkStarted();
+    return false;
+  }
+
+  @Override
+  public long getSizeInBytes() {
+    throw new UnsupportedOperationException("Method not supported");
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return false;
+  }
+
+  @Override
+  public void shutdown() throws StoreException {
+    started = false;
+  }
+
+  /**
+   * @return {@code true} if this store has been started successfully.
+   */
+  boolean isStarted() {
+    return started;
+  }
+
+  private void checkStarted() throws StoreException {
+    if (!started) {
+      throw new StoreException("Store not started", StoreErrorCodes.Store_Not_Started);
+    }
+  }
+
+  /**
+   * Detects duplicates in {@code writeSet}
+   * @param writeSet the {@link MessageWriteSet} to detect duplicates in
+   * @throws StoreException if a duplicate is detected
+   */
+  // TODO: move into MessageWriteSet?
+  private void checkDuplicates(MessageWriteSet writeSet) throws StoreException {
+    List<MessageInfo> infos = writeSet.getMessageSetInfo();
+    if (infos.size() > 1) {
+      Set<StoreKey> seenKeys = new HashSet<>();
+      for (MessageInfo info : infos) {
+        if (!seenKeys.add(info.getStoreKey())) {
+          throw new IllegalArgumentException("WriteSet contains duplicates. Duplicate detected: " + info.getStoreKey());
+        }
+      }
+    }
+  }
+
+  @Override
+  public String toString() {
+    return "PartitionId: " + partitionId.toPathString() + " in the cloud";
+  }
+
+  private class CloudWriter implements Write {
+    private final CloudBlobStore cloudBlobStore;
+    private final List<MessageInfo> messageInfoList;
+    private int messageIndex = 0;
+
+    CloudWriter(CloudBlobStore cloudBlobStore, List<MessageInfo> messageInfoList) {
+      this.cloudBlobStore = cloudBlobStore;
+      this.messageInfoList = messageInfoList;
+    }
+
+    @Override
+    public int appendFrom(ByteBuffer buffer) throws IOException {
+      return 0;
```

Comment:
no implementation?

Reply:
It's never called.  I'll change it to throw UnsupportedOperationException.